docs/sources/clients/promtail/logrotation.md

```diff
@@ -0,0 +1,137 @@
+
+We recommend (2) as that is the one which works well with Promtail (or any similar log scraping agent) without any data loss. Now let's understand how we exactly configure log rotation in different platforms.
+
+## Configure log rotation
+
+Your logs can be rotated by different components depending on where you are running your applications or services. If you are running them on Linux machines, high chance you will be using [`logrotate`](https://man7.org/linux/man-pages/man8/logrotate.8.html) utility. However, if you are running in Kubernetes, it's not that obvious who rotates the logs and interestingly it may depend on what container runtime your Kubernetes cluster is using.
+
+### Non-Kubernetes
+
+As mentioned above, in Linux machines, log rotation is often handled by [`logrotate`](https://man7.org/linux/man-pages/man8/logrotate.8.html) utility.
+
+It supports both types of log rotation explained before.
+
+#### Copy and Truncate
+```
+/var/log/apache2/*.log {
+        weekly
+        maxsize 1G
+        copytruncate
+}
+```
+
+Here `copytruncate` mode works exactly like (1) explained above.
+
+#### Rename and Create **(Recommend)**
+```
+/var/log/apache2/*.log {
+        weekly
+        maxsize 1G
+        create
+}
+```
+Here `create` mode works like (2) explained above. The `create` mode is optional because it's the default mode in `logrotate`.
+
+The configuration for `logrotate` is usually located in `/etc/logrotate/`.
+
+It has a wide range of [options](https://man7.org/linux/man-pages/man8/logrotate.8.html) for compression, mailing, running scripts pre- and post-rotation, etc.
+
+### Kubernetes
+
+[Kubernetes Service Discovery in Promtail]({{<relref "./scraping.md">}}#kubernetes-discovery) also uses file-based scraping. Meaning, logs from your pods are stored on the nodes and Promtail scrapes them from those files.
+
+We can [configure](https://kubernetes.io/docs/concepts/cluster-administration/logging/#log-rotation) the `kubelet` process running on each node to manage log rotation via two configs.
+
+1. `containerLogMaxSize` - It defines the maximum size of the container log file before it is rotated. For example: "5Mi" or "256Ki". Default: "10Mi".
+2. `containerLogMaxFiles` - It specifies the maximum number of container log files that can be present per container. Default: 5
+
+Both should be part of the `kubelet` config. If you run a managed version of Kubernetes in Cloud, refer to your cloud provider documentation for configuring `kubelet`. Examples [GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/node-system-config#create), [AKS](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#use-custom-node-configuration) and [EKS](https://eksctl.io/usage/customizing-the-kubelet/#customizing-kubelet-configuration).
+
+NOTE: Log rotation managed by `kubelet` supports only rename + create (for a good reason) and doesn't support copy + truncate.
```

Comment:
If you're not going to tell me the reason, I'd just delete "(For a good reason)" in this NOTE.

Reply:
The reason is what explained above. Why other method `copy  + truncate` is problematic

Reply:
Then just delete "(For a good reason)"

Reply:
Done.